File: server/apps/pages/api/serializers/testimonial.py
from rest_framework import serializers

from ...models import Testimonial
from ..mixins.translations import TranslationMixin


class TestimonialTranslationMixin(TranslationMixin):
    """Миксин для переводов полей testimonial"""

    def get_name(self, obj):
        return self.get_translated_field(obj, "name")

    def get_role(self, obj):
        return self.get_translated_field(obj, "role")

    def get_comment(self, obj):
        return self.get_translated_field(obj, "comment")


class TestimonialSerializer(TestimonialTranslationMixin, serializers.ModelSerializer):
    name = serializers.SerializerMethodField()
    role = serializers.SerializerMethodField()
    comment = serializers.SerializerMethodField()

    class Meta:
        model = Testimonial
        fields = (
            "id",
            "name",
            "role",
            "comment",
            "image",
            "rating",
            "is_active",
            "order",
        )

File: intro-app/src/api/modules/testimonial.js
import { publicApiClient } from "../client";
import { API_ENDPOINTS } from "../../constants/constants";

export const TestimonialApi = {
    async getAll() {
        try {
            const { data } = await publicApiClient.get(
                API_ENDPOINTS.modules.testimonial
            );

            if (data?.data && Array.isArray(data.data)) {
                return data.data.map(item => ({
                    id: item.id,
                    name: item.name,
                    role: item.role,
                    comment: item.comment,
                    image: item.image,
                    rating: item.rating,
                    isActive: item.is_active,
                    order: item.order,
                }));
            }

            return [];
        } catch (error) {
            throw error;
        }
    }
};
